avm/utl/types/avm-common-types/main.bicep

```diff
@@ -55,8 +57,8 @@ type diagnosticSettingFullType = {
   marketplacePartnerResourceId: string?
 }
 
-// Type with only metrics available
 @export()
+@description('An AVM-aligned type for a diagnostic setting. To be used if only metrics are supported by the resource provider.')
```

Comment:
I don't know why but this additional description made me wonder if discriminators may become handy for types which are mutually exclusive. For example the discriminator type for managedIdentity may be 'systemAssigned'/'userAssigned'/'systemUserAssigned'. I'd like to have a quick look at that

Reply:
Hey @eriqua,
discriminators rely on one property being the same across all types (the discriminator). Unfortunately, this is not the case for any of the UDTs as they don't differ in property value, but the properties that are available themselves. 
Of course we could add an artifictial value, but then a user of a module (e.g., KeyVault) would need to know which of the types to choose, each time. How it would be intended right now, the module owner would need to know that, once, when creating the module (/ implementing the UDT). The module user would then not have a choice and could only pass the values the module supports.

Reply:
I meant the artificial property of course,  added on top of existing properties exactly for that purpose. An additional "type" or "mode" that the user can leverage to then access only the properties they want. 
Hence why I wanted to investigate on it, mainly to always import the same "wrapper" type, including the others.
However as you correctly said, in that case the effort of choosing the right type based on the discriminator would be on the end user rather than on the module owners.
Approving.